### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR updates the following packages:

- `mesa`, update from version 21.1.4 to 21.2.5;
- `cmake`, update from version 3.20.5 to 3.22.0;
- `curl`, update from version 7.77.0 to 7.80.0;
- `freetype`, update from version 2.10.2 to 2.11.0;
- `nano`, update from version 5.8 to 5.9;
- `libuv`, update from version 1.41.1 to 1.42.0;
- `libxfont2`, update from version 2.0.4 to 2.0.5;
- `libxft`, update from version 2.3.3 to 2.3.4;
- `libjpeg-turbo`, update from version 2.1.0 to 2.1.2;
- `pango`, update from version 1.48.1 to 1.48.10;
- `libepoxy`, update from version 1.5.8 to 1.5.9;
- `harfbuzz`, update from version 2.8.2 to 3.1.2;
- `glib`, update from version 2.68.3 to 2.70.1;
- `libarchive`, update from version 3.5.1 to 3.5.2;
- `libgpg-error`, update from version 1.42 to 1.43;
- `libgcrypt`, update from version 1.8.5 to 1.9.4;
- `libsoup`, update from version 2.72.0 to 2.74.2;
- `wget`, update from version 1.21.1 to 1.21.2;
- `libxkbcommon`, update from version 1.3.0 to 1.3.1;
- `wayland-protocols`, update from version 1.21 to 1.24;
- `libinput`, update from version 1.16.4 to 1.19.2;
- `xorg-proto`, update from version 2021.4 to 2021.5;
- `libxi`, update from version 1.7.10 to 1.8;
- `glib-network`, update from version 2.68.2 to 2.70.0;
- `icu`, update from version 69.1 to 70.1;
- `libevdev`, update from version 1.9.1 to 1.12.0;
- `libdrm`, update from version 2.4.107 to 2.4.109;
- `fribidi`, update from version 1.0.10 to 1.0.11;
- `xkeyboard-config`, update from version 2.33 to 2.34;
- `sqlite`, update from version 3.36.0 to 3.37.0;
- `diffutils`, update from version 3.7 to 3.8;
- `grep`, update from version 3.5 to 3.7;
- `libffi`, update from version 3.3 to 3.4.2;
- `bc`, update from version 3.2.4 to 5.2.1;
- `openssl`, update from version 1.1.1k to 1.1.1l;
- `gdbm`, update from version 1.20 to 1.22;
- `gzip`, update from version 1.10 to 1.11;
- `iana-etc`, update from version 2021.06.11 to 2021.11.24;
- `coreutils`, update from version 8.32 to 9.0;
- `tzdata`, update from version 2021a to 2021e;
- `file`, update from version 5.40 to 5.41;
- `libpipeline`, update from version 1.5.2 to 1.5.4;
- `vim`, update from version 8.2.3001 to 8.2.3704;
- `bash`, update from version 5.1.8 to 5.1.12.

Furthermore, this PR fixes the following CVE's:
- Low: [CVE-2021-22924](https://nvd.nist.gov/vuln/detail/CVE-2021-22924#) in `curl`
- Medium: [CVE-2021-22922](https://nvd.nist.gov/vuln/detail/CVE-2021-22922#), [CVE-2021-22923](https://nvd.nist.gov/vuln/detail/CVE-2021-22923#), [CVE-2021-22925](https://nvd.nist.gov/vuln/detail/CVE-2021-22925#) and [CVE-2021-22947](https://nvd.nist.gov/vuln/detail/CVE-2021-22947#) in `curl` and [CVE-2021-3712](https://nvd.nist.gov/vuln/detail/CVE-2021-3712) in `openssl`.
- High: [CVE-2021-22926](https://nvd.nist.gov/vuln/detail/CVE-2021-22926#) and [CVE-2021-22946](https://nvd.nist.gov/vuln/detail/CVE-2021-22946#) in `curl`, [CVE-2021-33560](https://nvd.nist.gov/vuln/detail/CVE-2021-33560) in `libgcrypt` and [CVE-2021-3711]https://nvd.nist.gov/vuln/detail/CVE-2021-3711) in `openssl`.
- Critical: [CVE-2021-22945](https://nvd.nist.gov/vuln/detail/CVE-2021-22945#) in `curl`.

One notoriously broken port finally got fixed, glad to see `pango` on this list. It seems that the `icu` issues from #123 are fixed, but the full rebuild might show some more of them. I've excluded `gawk` (autoconf 2.71 required, awaiting config patches upstream), `python` (rework the patch again), `gdk-pixbuf` (`gdk-pixbuf-xlib` got split off), `shared-mime-info` (needs `xmlto`, which pulls in texlive in the docker image) and `xorg-server` (xwayland got split off, needs investigating) from the updates, for the reasons mentioned. Furthermore, `llvm` and `gcc` are blocked on codegen issues, tho `gcc` will receive an update soon for unrelated reasons. It is this reason, that I didn't touch `binutils`. I want to tackle all of them in a separate toolchain upgrade PR.

Blocked on managarm/mlibc#336 (merged) and managarm/mlibc#337 (still open)

### CODE REVIEW

**Review by avdgrinten — COMMENTED** [review]
I don't like that this patches out `fsync`. If `fsync` failures are an issue, why don't we just make `fsync` return success in all cases in mlibc (+ print a warning)?

**Comment by Dennisbonke** [comment]
> I don't like that this patches out `fsync`. If `fsync` failures are an issue, why don't we just make `fsync` return success in all cases in mlibc (+ print a warning)?

Hmm yeah I agree. I too would like to drop this patch. Luckily, managarm/mlibc#307 does this for me, and is only pending the `pthread_cond_timedwait()` fixes (I have the rest locally addressed, just need to fix up and force push). I’ll post some more info regarding pthread somewhere today in discord as qookie requested, block this PR on that one, and drop the offending patch from nano.

**Review by no92 — COMMENTED** [review]
Very minor nitpick: the libdrm version in the commit message is off-by-one.

The same bump is also included in #153, whatever gets merged second should drop this.

File: bootstrap.d/app-editors.yml [hunk 1]
```diff
@@ -5,12 +5,14 @@ packages:
...
       git: 'https://git.savannah.gnu.org/git/nano.git'
-      tag: 'v5.8'
-      version: '5.8'
+      tag: 'v5.9'
+      version: '5.9'
       tools_required:
         - host-autoconf-v2.69
         - host-automake-v1.15
         - host-pkg-config
+        - virtual: pkgconfig-for-target
```
Comment (no92) [inline bootstrap.d/app-editors.yml:14]: As this seems uncommon, the question for me is whether adding this is necessary?

**Pull request merged**